backend/spotihue/oauth.py

```diff
@@ -1,23 +1,34 @@
 import logging
+import os
 import webbrowser
-from urllib import parse
 from typing import Optional
+from urllib import parse
 
 from spotipy import oauth2, util
 
 
 logger = logging.getLogger(__name__)
 
+SPOTIFY_OAUTH_TIMEOUT = os.getenv('SPOTIFY_OAUTH_TIMEOUT', 60 * 3)  # default 3 minutes
+
+
+class HTTPServer(oauth2.HTTPServer):
+    timeout = SPOTIFY_OAUTH_TIMEOUT
+
 
 def start_local_http_server(port, handler=oauth2.RequestHandler):
-    server = oauth2.HTTPServer(("0.0.0.0", port), handler)
+    server = HTTPServer(("0.0.0.0", port), handler)
     server.allow_reuse_address = True
     server.auth_code = None
     server.auth_token_form = None
     server.error = None
     return server
 
 
+class SpotifyOauthSocketTimeout(oauth2.SpotifyOauthError):
```

Comment:
Is this necessary? Just a pass here?

Reply:
It's not strictly necessary. I was making a distinction between a task erroring out because of a Spotify auth problem, vs. a task erroring out because it waited for a user to type in their info and they never did. Whether or not it's desired behavior to distinguish between those two cases, is debatable!
A oauth2.SpotifyOauthError exception is raised if there's been a spotipy framework error connecting/authenticating to Spotify. This little SpotifyOauthSocketTimeout subclass is raised by our extended Oauth logic if the user fails to interact with Spotify at all (ie. if no auth code OR error message is returned from Spotify). And, this SpotifyOauthSocketTimeout subclass is passed as the Spotify setup task's `throws` argument. The [throws argument](https://docs.celeryq.dev/en/latest/reference/celery.app.task.html#celery.app.task.Task.throws) to a celery Task object denotes exceptions that may get thrown, but that shouldn't be considered "erroneous" behavior.